### PR TITLE
chore(deps): Bump Go dependencies

### PR DESCRIPTION
**Pull request by Techassi**
This fixes the critical vulnerability CVE-2024-41110 in `github.com/docker/docker`.

```[tasklist]
# Reviewer
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

### CODE REVIEW

File: rust/stackablectl/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Changed
+
+- helm-sys: Bump Go dependencies to fix critical vulnerability in
+  `github.com/docker/docker`, See [CVE-2024-41110] ([#xxx]).
+
+[#xxx]: https://github.com/stackabletech/stackable-cockpit/pull/xxx
```
Comment (sbernauer) [inline rust/stackablectl/CHANGELOG.md]:
```suggestion
  `github.com/docker/docker`, See [CVE-2024-41110] ([#313]).

[#313]: https://github.com/stackabletech/stackable-cockpit/pull/313
```

**Review by sbernauer — CHANGES_REQUESTED** [review]
LGTM otherwise

**Review by sbernauer — APPROVED** [review]
Merging this, as @Techassi is on vacation

**Comment by NickLarsenNZ** [comment]
Gomod2nix needed updating via `make regenerate-nix`

**Comment by sbernauer** [comment]
Thanks, created https://github.com/stackabletech/stackable-cockpit/pull/314 for that

**Pull request merged**